src/scf/mixing.jl

```diff
@@ -3,6 +3,15 @@ using IterativeSolvers
 using Statistics
 import Base: @kwdef
 
+abstract type Mixing end
+
+# Approximation to the dielectric matrix, which is diagonal
```

Comment:
What does it mean diagonal in this sense?

Reply:
Why not just get rid of this MixingDiagonal type, have functions mix_potential and mix_density, and define `mix_potential(mixing::Mixing, basis, deltaF) = mix_density(mixing::Mixing, basis, deltaF)`? That way by default it's the same, but mixing rules can overwrite it if needed.

Reply:
I thought about that, but the fact that this works is really a special case for these types of dielectric models. Ok arguably "diagonal" is probably not the best term for it, but I thought it would be better to require an active opt-in into this equality rather than an opt-out. Don't you think?

Reply:
Most preconditioners are really very crude things that don't care so much whether it's potentials or densities, so I think it makes sense to have the simplest possible API and let more complicated algorithms overwrite when needed

Reply:
Ok. If it shoots us in the back some day, I'll point at you ;)

Reply:
deal